### PR TITLE
refactor: flake8 resolution for cobra.Solution

### PR DESCRIPTION
**Pull request by synchon**
* [ ] fix #(issue number)
* [x] description of feature/fix
* [x] tests added/passed
* [ ] add an entry to the [next release](../release-notes/next-release.md)

This PR resolves `flake8` issues for `cobra/core/solution.py` and `test_core/test_solution.py` and also adds type annotations and docstrings. Upgrades code style wherever required.

### CODE REVIEW

File: src/cobra/core/solution.py [hunk 1]
```diff
@@ -1,32 +1,42 @@
...
-class Solution(object):
+class Solution:
     """
     A unified interface to a `cobra.Model` optimization solution.
 
-    Notes
-    -----
-    Solution is meant to be constructed by `get_solution` please look at that
-    function to fully understand the `Solution` class.
+    Parameters
```
Comment (Midnighter) [inline src/cobra/core/solution.py:26]: Parameters are for the `__init__` method. I'm not sure why you move them here?

File: src/cobra/core/solution.py [hunk 2]
```diff
@@ -40,78 +50,97 @@ class Solution(object):
...
-        status,
-        fluxes,
-        reduced_costs=None,
-        shadow_prices=None,
-        **kwargs
-    ):
+        objective_value: float,
+        status: str,
+        fluxes: pd.Series,
+        reduced_costs: pd.Series = None,
```
Comment (Midnighter) [inline src/cobra/core/solution.py]:
```suggestion
        reduced_costs: Optional[pd.Series] = None,
```

File: src/cobra/core/solution.py [hunk 3]
```diff
@@ -40,78 +50,97 @@ class Solution(object):
...
+        shadow_prices: pd.Series = None,
```
Comment (Midnighter) [inline src/cobra/core/solution.py]:
```suggestion
        shadow_prices: Optional[pd.Series] = None,
```

File: src/cobra/core/solution.py [hunk 4]
```diff
@@ -40,78 +50,97 @@ class Solution(object):
...
         self.shadow_prices = shadow_prices
 
-    def __repr__(self):
-        """String representation of the solution instance."""
+    def __repr__(self) -> str:
+        """Return a string representation of the solution instance."""
         if self.status != OPTIMAL:
-            return "<Solution {0:s} at 0x{1:x}>".format(self.status, id(self))
-        return "<Solution {0:.3f} at 0x{1:x}>".format(self.objective_value, id(self))
+            return f"<Solution {self.status} at 0x{id(self):x}>"
```
Comment (Midnighter) [inline src/cobra/core/solution.py]:
```suggestion
            return f"<Solution {self.status} at {id(self):#x}>"
```

File: src/cobra/core/solution.py [hunk 5]
```diff
@@ -40,78 +50,97 @@ class Solution(object):
...
+        return f"<Solution {self.objective_value:.3f} at 0x{id(self):x}>"
```
Comment (Midnighter) [inline src/cobra/core/solution.py]:
```suggestion
        return f"<Solution {self.objective_value:.3f} at {id(self):#x}>"
```

File: src/cobra/core/solution.py [hunk 6]
```diff
@@ -40,78 +50,97 @@ class Solution(object):
...
+        return pd.DataFrame(
+            {"fluxes": self.fluxes, "reduced_costs": self.reduced_costs}
+        )
 
 
-def get_solution(model, reactions=None, metabolites=None, raise_error=False):
+def get_solution(
+    model: "Model",
+    reactions: Iterable["Reaction"] = None,
+    metabolites: Iterable["Metabolite"] = None,
```
Comment (Midnighter) [inline src/cobra/core/solution.py]: Please add `Optional`

**Review by Midnighter — CHANGES_REQUESTED** [review]
Few minor comments, otherwise looks good.

File: src/cobra/core/solution.py (hunk 1 above)
Comment (synchon) [reply src/cobra/core/solution.py:26]: From [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes), I understand that the parameters for `__init__` need to be documented at the class level.

File: src/cobra/core/solution.py (hunk 2 above)
Comment (synchon) [reply src/cobra/core/solution.py]: Ah yes, missed that.

File: src/cobra/core/solution.py (hunk 6 above)
Comment (synchon) [reply src/cobra/core/solution.py]: My bad, again missed those.

File: src/cobra/core/solution.py (hunk 1 above)
Comment (Midnighter) [reply src/cobra/core/solution.py:26]: I see 👍🏼

**Review by Midnighter — APPROVED** [review]

**Pull request merged**